File: parameter_object_and_result_object_pattern/mix_exercises/demo1/app1.ts
class UserParams {
  name: string;
  age: number;
  address: string;
  email: string;
  phone: string;
  username: string;

  constructor(
    name: string,
    age: number,
    address: string,
    email: string,
    phone: string,
    username: string
  ) {
    this.name = name;
    this.age = age;
    this.address = address;
    this.email = email;
    this.phone = phone;
    this.username = username;
  }
}

class UserParams2 {
  constructor(
    public name: string,
    public age: number,
    public address: string,
    public email: string,
    public phone: string,
    public username: string
  ) {}
}

File: parameter_object_and_result_object_pattern/mix_exercises/demo1/app1.py
from dataclasses import dataclass


class UserParams:
    def __init__(self, name, age, address, email, phone, user_name):
        self.name = name
        self.age = age
        self.address = address
        self.email = email
        self.phone = phone
        self.user_name = user_name


class UserParams2:
    def __init__(
        self,
        name: str,
        age: int,
        address: str,
        email: str,
        phone: str,
        user_name: str = None,
    ):
        self.name = name
        self.age = age
        self.address = address
        self.email = email
        self.phone = phone
        self.user_name = user_name
        
@dataclass
class UserParams3:
    name: str
    age: int
    address: str
    emai: str
    phone: str
    user_name: str = None)
    


class UserParams4:


user1 = UserParams("John", 30, "123 Main St", "", "555-1234", "john_doe")
user2 = UserParams2("John", 30, "123 Main St", "", "555-1234", "john_doe")
user3 = UserParams3("John", 30, "123 Main St", "", "555-1234", "john_doe")
